### PR TITLE
Makes ammo stackers upgradable via matter bins

### PR DESCRIPTION
**Pull request by Omppusolttu**
## About The Pull Request

Ammo stacker max capacity can now be upgraded via matter bins:
T1 = 12
T2 = 15
T3 = 18
T4 = 21

The ammo stacker now contains 3 matter bins.

## Why It's Good For The Game

Ammo stackers are not used for their intended purpose of holding high explosive materials atleast in the case of gunpowder due to their incredibly low capacity and clunkiness in usage, especially when compared to the abundant and easy to use crates.

## Changelog
:cl:
add: Matter bins to Ammo stackers
/:cl:

### CODE REVIEW

**Review by Karmic-Skink — CHANGES_REQUESTED** [review]
Those numbers are way too high.

**Comment by Bokkiewokkie** [comment]
@Omppusolttu I think it'd be better if you changed the amount increase per tier instead of the default capacity instead of doing what you did now.

File: nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/automation.dm [hunk 1]
```diff
@@ -208,6 +208,7 @@
 
 /obj/item/circuitboard/machine/ammo_sorter
 	name = "Ammo sorter (circuitboard)"
+	req_components = list(/obj/item/stock_parts/matter_bin = 1)
```
Comment (Karmic-Skink) [inline nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/automation.dm:211]: Given the amount this is holding, it should be more matter bins.

File: nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/automation.dm [hunk 2]
```diff
@@ -287,7 +288,7 @@
 	anchored = TRUE
 	var/id = null
 	var/list/loaded = list() //What's loaded in?
-	var/max_capacity = 12 //Max cap for holding.
+	var/max_capacity = 10	//Max cap for holding.
```
Comment (Karmic-Skink) [inline nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/automation.dm:291]: Reducing this number is going to cause havoc where it is already mapped in.

File: nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/automation.dm [hunk 3]
```diff
@@ -336,6 +337,10 @@
 		for(var/obj/item/C in loaded)
 			. += "<br/><span class='notice'>[C].</span>"
 
+/obj/machinery/ammo_sorter/RefreshParts()
+	for(var/obj/item/stock_parts/matter_bin/MB in component_parts)
+		max_capacity = MB.rating*10
```
Comment (Karmic-Skink) [inline nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/automation.dm:342]: Increasing by 100% of the base value per tier is way too over the top.

**Review by Karmic-Skink — CHANGES_REQUESTED** [review]
Unbalanced as heck.

File: nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/automation.dm [hunk 4]
```diff
@@ -336,6 +337,10 @@
 		for(var/obj/item/C in loaded)
 			. += "<br/><span class='notice'>[C].</span>"
 
+/obj/machinery/ammo_sorter/RefreshParts()
```
Comment (Karmic-Skink) [inline nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/automation.dm:340]: This block isn't taking into account that you've changed it to use three bins.

**Review by Karmic-Skink — CHANGES_REQUESTED** [review]

File: nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/automation.dm [hunk 5]
```diff
@@ -336,6 +337,11 @@
...
+	max_capacity = 0
+	for(var/obj/item/stock_parts/matter_bin/MB in component_parts)
+		max_capacity += MB.rating*1+3
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/automation.dm:343]:
```suggestion
		max_capacity += MB.rating+3
```
x*1 = x

**Review by Bokkiewokkie — APPROVED** [review]

**Pull request merged**